api/requests/views.py

```diff
@@ -118,3 +121,53 @@ class PreprintRequestDetail(JSONAPIBaseView, generics.RetrieveAPIView, PreprintR
 
     def get_object(self):
         return self.get_request()
+
+class RequestActionList(JSONAPIBaseView, generics.ListAPIView):
+    permission_classes = (
+        permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+    )
+
+    required_read_scopes = [CoreScopes.ACTIONS_READ]
+    required_write_scopes = [CoreScopes.NULL]
+
+    view_category = 'requests'
+    view_name = 'request-action-list'
+
+    def get(self, request, *args, **kwargs):
+        request_id = self.kwargs['request_id']
+        if PreprintRequest.objects.filter(_id=request_id).exists():
+            return PreprintRequestActionList.as_view()(request, *args, **kwargs)
+        else:
+            raise NotFound
+
+class PreprintRequestActionList(JSONAPIBaseView, generics.ListAPIView, PreprintRequestMixin, ListFilterMixin):
+    permission_classes = (
+        permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+        MustBeModerator,
```

Comment:
Same as other PR, do these perm classes work on this list view properly or does it need to be handled by the qs. I'm betting based on your tests, this actually works.

Reply:
Permissions classes use `has_object_permission` for individual objects, and `has_permission` for views in general. `MustBeModerator` uses `has_permission` to get the provider related to the request, and checks perms based on that.